mne/viz/epochs.py

```diff
@@ -942,26 +961,72 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, tmin=None, tmax=None,
+                # std across channels
+                psd_std = np.std(np.mean(psds, axis=0), axis=0)
+                hyp_limits = (psd_mean - psd_std, psd_mean + psd_std)
+            elif area_mode == 'range':
+                hyp_limits = (np.min(np.mean(psds, axis=0), axis=0),
+                              np.max(np.mean(psds, axis=0), axis=0))
+            else:  # area_mode is None
+                hyp_limits = None
+
+            ax.plot(freqs, psd_mean, color=color)
+            if hyp_limits is not None:
+                ax.fill_between(freqs, hyp_limits[0], y2=hyp_limits[1],
+                                color=color, alpha=area_alpha)
+        else:
+            psd_list.append(psds)
         if make_label:
             if ii == len(picks_list) - 1:
                 ax.set_xlabel('Frequency (Hz)')
-            ax.set(ylabel=ylabel, title=title, xlim=(freqs[0], freqs[-1]))
+            ax.set_ylabel(ylabel)
+            ax.set_title(titles_list[ii])
+            ax.set_xlim(freqs[0], freqs[-1])
+
+        ylabels.append(ylabel)
+
+    for key, ls in zip(['lowpass', 'highpass', 'line_freq'],
+                       ['--', '--', '-.']):
+        if epochs.info[key] is not None:
+            for ax in ax_list:
+                ax.axvline(epochs.info[key], color='k', linestyle=ls,
+                           alpha=0.25, linewidth=2, zorder=2)
+
+    if not average:
+        picks = np.concatenate(picks_list)
+        psd_list = np.concatenate(psd_list)
+        types = np.array([channel_type(epochs.info, idx) for idx in picks])
+        # Needed because the data do not match the info anymore.
+        info = create_info([epochs.ch_names[p] for p in picks], epochs.info['sfreq'],
+                           types)
+        info['chs'] = [epochs.info['chs'][p] for p in picks]
+        valid_channel_types = ['mag', 'grad', 'eeg', 'seeg', 'eog', 'ecg',
+                               'emg', 'dipole', 'gof', 'bio', 'ecog', 'hbo',
+                               'hbr', 'misc']
+        ch_types_used = list()
+        for this_type in valid_channel_types:
+            if this_type in types:
+                ch_types_used.append(this_type)
+        assert len(ch_types_used) == len(ax_list)
+        unit = ''
+        units = {t: yl for t, yl in zip(ch_types_used, ylabels)}
+        titles = {c: t for c, t in zip(ch_types_used, titles_list)}
+        picks = np.arange(len(psd_list))
+        if not spatial_colors:
+            spatial_colors = color
+        _plot_lines(psd_list, info, picks, fig, ax_list, spatial_colors,
+                    unit, units=units, scalings=None, hline=None, gfp=False,
+                    types=types, zorder='std', xlim=(freqs[0], freqs[-1]),
+                    ylim=None, times=freqs, bad_ch_idx=[], titles=titles,
+                    ch_types_used=ch_types_used, selectable=True, psd=True,
+                    line_alpha=line_alpha, nave=None)
+    for ax in ax_list:
+        ax.grid(True, linestyle=':')
```

Comment:
any chance to factorize this code to reuse the raw.plot_psd code? it must be very similar

Reply:
Yes, I'll make an attempt. Where should it go? viz/utils.py?

Reply:
Is there a reason why Raw.plot_psd uses Welch, and Epochs.plot_psd uses multitaper? Is this something we should parametrize for the user?

Reply:
> Is there a reason why Raw.plot_psd uses Welch, and Epochs.plot_psd uses multitaper? Is this something we should parametrize for the user?

Yes. Welch chunks the data in time, computes a bunch of FFTs, takes magnitudes, then averages. Epochs are already "chunked in time" so no need for a Welch operation. Might as well use multitaper (which uses multiple window functions on a given single time segment) to improve the spectral estimation.

Can you add an explanation for this difference in `epochs.plot_psd`?